ts/keys.go

```diff
@@ -78,31 +78,33 @@ func MakeDataKey(name string, source string, r Resolution, timestamp int64) prot
 }
 
 // DecodeDataKey decodes a time series key into its components.
-func DecodeDataKey(key proto.Key) (string, string, Resolution, int64) {
-	var (
-		name          []byte
-		source        []byte
-		resolutionInt int64
-		timeslot      int64
-		remainder     = key
-	)
-
+func DecodeDataKey(key proto.Key) (string, string, Resolution, int64, error) {
```

Comment:
Maybe, though named return parameters have their own downsides. I'm just passing by this code and will let @mrtracy fix this up in a subsequent PR if he wishes.